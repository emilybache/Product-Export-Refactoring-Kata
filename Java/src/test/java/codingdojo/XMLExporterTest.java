package codingdojo;

import org.approvaltests.Approvals;
import org.approvaltests.core.Options;
import org.approvaltests.scrubbers.DateScrubber;
import org.approvaltests.scrubbers.RegExScrubber;

import java.util.Arrays;
import java.util.Collection;

import static codingdojo.SampleModelObjects.FlagshipStore;
import static codingdojo.SampleModelObjects.OldOrder;
import static codingdojo.SampleModelObjects.RecentOrder;

import org.junit.jupiter.api.Test;

public class XMLExporterTest {

    @Test
    public void exportFull() throws Exception {
        Collection<Order> orders = Arrays.asList(RecentOrder, OldOrder);
        String xml = XMLExporter.exportFull(orders);
        Approvals.verifyXml(xml);
    }

    @Test
    public void exportTaxDetails() throws Exception {
        Collection<Order> orders = Arrays.asList(RecentOrder, OldOrder);
        String xml = XMLExporter.exportTaxDetails(orders);
        Approvals.verifyXml(xml);
    }

    @Test
    public void exportStore() throws Exception {
        Store store = FlagshipStore;
        String xml = XMLExporter.exportStore(store);
        Approvals.verifyXml(xml);
    }

    @Test
    public void exportHistory() throws Exception {
        Collection<Order> orders = Arrays.asList(RecentOrder, OldOrder);
        String xml = XMLExporter.exportHistory(orders);
        String regex = "createdAt='[^']+'";
        Approvals.verifyXml(xml, new Options().withScrubber(
                DateScrubber.getScrubberFor("2022-06-21T11:29Z")
        ));
    }
}
