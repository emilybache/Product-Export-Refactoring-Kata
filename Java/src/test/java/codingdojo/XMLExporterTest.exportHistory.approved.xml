<?xml version="1.0" encoding="UTF-8"?><orderHistory createdAt="[Date1]">
  <order date="[Date2]" totalDollars="149.99">
    <product id="EVENT02">Makeover</product>
  </order>
  <order date="[Date3]" totalDollars="14.99">
    <product id="LIPSTICK01">Cherry Bloom</product>
  </order>
</orderHistory>
